### PR TITLE
New translations system

### PR DESCRIPTION
**Pull request by benalleng**
Works towards closing #152.

I have some questions on how to best plan for the translations moving forward
1. How should reused translations like a copy or continue button be handled?
2. Should there be a set template to ensure easy translation changes so that each translation page can be quickly replicated.
I think making a simple translation key template would help with 2 and I went ahead and listed them below.
```
const translations = {
    ${route}: {
        ${lang}: {
            title: ‘Lorem ipsum’,
            heading: ‘Lorem ipsum’,
            subheading_one: ‘Lorem ipsum’,
            paragraph_one: ‘Lorem ipsum dolor sit amet, ex ea commodo consequat.’,
            link: ‘Click Me!’,
            paragraph_two: ‘Lorem ipsum dolor sit amet, ex ea commodo consequat.’,
            ${component_object_param}: ‘Error Message’,
            next: ‘Continue’,
            ...
            paragraph_nine: ‘Lorem ipsum dolor sit amet, ex ea commodo consequat.’,,
            paragraph_(n): ‘Lorem ipsum dolor sit amet, ex ea commodo consequat.’,
        },
    },
}
```
Most notably I think having some syntax-ordering i.e. not having all paragraphs together but ordering them how they appear on the page can help with quickly showing what work needs to be completed in the translations

### CODE REVIEW

**Comment by afterburn** [comment]
1. I suggest we create a translations file at the root of our project that we can use to define translations that are commonly reused. Something along the lines of: 
```ts
export const translations = {
  common: {
    en: {
      continue: 'Continue'
    }
  }
}
```
This file would then have to be imported in `lib/translate.tsx` as well, at least until we implement an auto-import translations solution.

2. I mean yeah that is how all existing translation files have been set up so far. To me it makes sense to continue with that approach. Also means we don't have to support other translation structures in `lib/translate.tsx` . I'm referring to these top-level ones:
```ts
export const translations = {
  [namespace]: {
    [lang]: {
      ...
    }
  }
}
```
As for how we name the translations keys that go into `...`, I don't really care as long as it's `snake_case` and makes at least some sense 😁.

**Review by afterburn — CHANGES_REQUESTED** [review]
Code looks good but noticed this issue in the Topbar/Navbar:

image

It should say: "Chapter 1 - Challenge 2"

File: lib/translate.tsx [hunk 1]
```diff
@@ -75,7 +71,9 @@ export function t(key: string, lang: string) {
   let translation = get(translations, `${lang}.${key}`)
 
   if (!translation) {
-    return `{${key}}`
+    // console.log(`No translation for {${lang}.${key}}`)
+    let fallbackTranslation = get(translations, `en.${key}`)
```
Comment (afterburn) [inline lib/translate.tsx:75]:
```js
// Fallback translation
return get(translations, `en.${key}`)
```

**Comment by afterburn** [comment]
@ecurrencyhodler feel free to merge this once it has your tACK

**Comment by ecurrencyhodler** [comment]
tACK!

**Pull request merged**